### PR TITLE
Add ability to remove faulted subscription

### PR DESCRIPTION
**Pull request by xiazen**
regards to #2968, add ability to remove faulted subscription from pubsub

### CODE REVIEW

**Comment by xiazen** [comment]
@dotnet-bot test netfx-functional please

File: src/OrleansRuntime/Streams/PubSub/PubSubRendezvousGrain.cs [hunk 1]
```diff
@@ -281,7 +281,8 @@ private void RemoveProducer(PubSubPublisherState producer)
         {
             counterConsumersRemoved.Increment();
             if (State.Consumers.Any(c => c.IsFaulted && c.Equals(subscriptionId)))
-                throw new FaultedSubscriptionException(subscriptionId, streamId);
+                logger.Warn(ErrorCode.Stream_RemoveFaultedSubscription,
```
Comment (jason-bragg) [inline src/OrleansRuntime/Streams/PubSub/PubSubRendezvousGrain.cs]: Originally faulted subscriptions were intended to be permanent, but it should probably be ok to remove them via the subscription management interfaces or by unsubscribing.  That being the case, I don't think we even need to check (and log) in this case.  So we can just remove this entire if block.

Comment (xiazen) [inline src/OrleansRuntime/Streams/PubSub/PubSubRendezvousGrain.cs]: yes boss

**Comment by jason-bragg** [comment]
@dotnet-bot test netfx-bvt  netstandard-win-bvt  please

**Pull request merged**